### PR TITLE
fix generate_range function in datetimes.py file

### PR DESCRIPTION
**Pull request by zky001**
change some issues be metioned at #35342

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by zky001** [comment]
I test all the cases on my machine but there still have some errors like `pytz.exceptions.NonExistentTimeError: 1907-01-01 00:00:00`
I want to know whether should I change NonExistentTimeError detect ways.

**Comment by zky001** [comment]
I don't know weather should the time Zone of Indian/Chagos show time like `1907-01-01 00:00:00` should be droped？

**Review by mroeschke — CHANGES_REQUESTED** [review]
This would need a test.

Also I don't think this is an appropriate place for the fix. I imagine the bug is in `generate_range`

Could you give this PR a more informative title as well? Thanks.

**Comment by zky001** [comment]
> This would need a test.
> 
> Also I don't think this is an appropriate place for the fix. I imagine the bug is in `generate_range`
> 
> Could you give this PR a more informative title as well? Thanks.

I have make some tests and they all passed,I will let PR have  a more informative title。
image

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -2398,6 +2398,7 @@ def generate_range(start=None, end=None, periods=None, offset=BDay()):
 
     elif end and not offset.is_on_offset(end):
         end = offset.rollback(end)
+        end = Timestamp(str(end).split(' ')[0] + ' 23:59:59')
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:2401]: I think this fix is too specific. The bug probably lies in not checking the end bounds

**Comment by mroeschke** [comment]
This PR still needs a test.

**Comment by simonjayhawkins** [comment]
@zky001 can you address https://github.com/pandas-dev/pandas/pull/35348#issuecomment-663341641 and https://github.com/pandas-dev/pandas/pull/35348#pullrequestreview-454628733

**Comment by jbrockmendel** [comment]
See my comment in #35342 on the appropriate fix

**Comment by dsaxton** [comment]
Closing as stale, @zky001 let us know if you'd still like to work on this

**Pull request closed without merging**